### Dataset Prediksi Harga Rumah


###### Dataset ini berisikan 50 data yang terkait tentang harga rumah, yang meliputi lokasi,tahun dibuat,kondisi,jumlah kamar,jumlah lantai dan garasi.Pada analisis ini ,terdapat dua metode regresi korelasi yaitu berganda dan sederhana.

###### [Sumber data](https://www.kaggle.com/datasets/zafarali27/house-price-prediction-dataset)

#### Import data dan membuat data frame

In [3]:
import pandas as pd
import openpyxl as op

data = pd.read_excel(r"C:\Users\Michael Zidane\Downloads\matana project\Harga rumah.xlsx")
df = pd.DataFrame(data)

print(df)

    Id  Area  Bedrooms  Bathrooms  Floors  YearBuilt  Location  Condition  \
0    1  1360         5          4       3       1970  Downtown  Excellent   
1    2  4272         5          4       3       1958  Downtown  Excellent   
2    3  3592         2          2       3       1938  Downtown       Good   
3    4   966         4          2       2       1902  Suburban       Fair   
4    5  4926         1          4       2       1975  Downtown       Fair   
5    6  3944         1          2       1       1906     Urban       Poor   
6    7  3671         1          1       2       1948     Rural       Poor   
7    8  3419         2          4       1       1925  Suburban       Good   
8    9   630         2          2       1       1932     Rural       Poor   
9   10  2185         3          3       1       2000  Downtown       Poor   
10  11  1269         1          4       2       1947  Suburban       Poor   
11  12  2891         4          4       3       1978     Urban  Excellent   

#### Regresi Linear Berganda

In [7]:
import pandas as pd
import statsmodels.api as sm

df_numerik = df.select_dtypes(include=['number'])
Y = df_numerik['Price']  
X = df_numerik.drop('Price', axis=1)
X = sm.add_constant(X)
model = sm.OLS(Y, X).fit()

print(model.summary())

                            OLS Regression Results                            
Dep. Variable:                  Price   R-squared:                       0.232
Model:                            OLS   Adj. R-squared:                  0.125
Method:                 Least Squares   F-statistic:                     2.165
Date:                Sun, 29 Mar 2026   Prob (F-statistic):             0.0652
Time:                        16:46:36   Log-Likelihood:                -691.55
No. Observations:                  50   AIC:                             1397.
Df Residuals:                      43   BIC:                             1410.
Df Model:                           6                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const        4.42e+06   2.27e+06      1.944      0.0

#### Cek asumsi Regresi Linear Berganda

Uji nomralitas residual

In [9]:
from scipy.stats import shapiro

residual = model.resid

stat, p = shapiro(residual)
print("p-value:", p)

p-value: 0.6450591080781855


###### karena p-value lebih dari 0,05  berarti data tersebut aman

Uji Multikolinearitas (VIF)

In [11]:
from statsmodels.stats.outliers_influence import variance_inflation_factor

vif = pd.DataFrame()
vif["VIF"] = [variance_inflation_factor(X.values, i) for i in range(X.shape[1])]
vif["Variabel"] = X.columns
vif

,VIF,Variabel
0,3682.699515,const
1,1.159999,Id
2,1.065000,Area
3,1.091707,Bedrooms
4,1.066099,Bathrooms
5,1.075906,Floors
6,1.138180,YearBuilt


###### karena VIF kurang dari 10 data tersebut aman

Uji Heteroskedastisitas

In [12]:
from statsmodels.stats.diagnostic import het_breuschpagan

test = het_breuschpagan(residual, X)
p_value = test[1]

print("p-value:", p_value)

p-value: 0.9816804630324065


###### karena p value lebih dari 0,05 maka data tersebut homokedastis

Uji autokorelasi

###### Durbin-Watson = 2.397 (~2 →  tidak ada autokorelasi )


#### Regresi linear sederhana

In [13]:
X1 = df_numerik[['Area']]
X1 = sm.add_constant(X1)

model_sederhana = sm.OLS(Y, X1).fit()
model_sederhana.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                  Price   R-squared:                       0.000
Model:                            OLS   Adj. R-squared:                 -0.020
Method:                 Least Squares   F-statistic:                   0.02116
Date:                Sun, 29 Mar 2026   Prob (F-statistic):              0.885
Time:                        17:11:08   Log-Likelihood:                -698.13
No. Observations:                  50   AIC:                             1400.
Df Residuals:                      48   BIC:                             1404.
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const       5.855e+05   9.86e+04      5.938      0.000    3.87e+05    7.84e+05
Area          -4.9853     34.270     -0.145      0.885     -73.890      63.920
==============================================================================
Omnibus:                       16.690   Durbin-Watson:                   2.418
Prob(Omnibus):                  0.000   Jarque-Bera (JB):                3.538
Skew:                          -0.106   Prob(JB):                        0.170
Kurtosis:                       1.714   Cond. No.                     7.01e+03
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The condition number is large, 7.01e+03. This might indicate that there are
strong multicollinearity or other numerical problems.
"""

#### Penjelasan metode regresi linear sederhana

###### Model regresi linear sederhana yang diperoleh adalah:

###### Price=585500−4.9853×Area

###### Model ini menunjukkan bahwa variabel Area memiliki hubungan negatif terhadap Price, di mana setiap kenaikan satu satuan luas area akan menurunkan harga sebesar 4.9853 satuan. Namun, berdasarkan hasil uji statistik, nilai p-value untuk variabel Area sebesar 0.885 (> 0.05), sehingga dapat disimpulkan bahwa variabel Area tidak berpengaruh signifikan terhadap Price.

###### Dengan demikian, model regresi linear sederhana ini belum mampu menjelaskan hubungan yang berarti antara Area dan Price.

#### Adjusted R-Square dan Interpretasinya

###### Nilai Adjusted R-Squared yang diperoleh adalah sebesar -0.020.

###### Nilai ini menunjukkan bahwa model regresi yang digunakan tidak mampu menjelaskan variasi dalam variabel Price. Bahkan, nilai Adjusted R-Squared yang negatif mengindikasikan bahwa model yang dibangun lebih buruk dibandingkan dengan model tanpa variabel independen (hanya menggunakan rata-rata).

###### Dengan demikian, dapat disimpulkan bahwa variabel Area tidak cukup kuat untuk menjelaskan perubahan pada Price, sehingga diperlukan variabel lain untuk membangun model yang lebih baik.